components/ingest-service/backend/elastic/actions.go

```diff
@@ -31,6 +31,43 @@ func (es *Backend) CreateBulkActionRequest(action backend.InternalChefAction) el
 	return es.createBulkUpdateRequestToTimeseriesIndex(mapping, time, action)
 }
 
+func (es *Backend) DeleteActionProjectTag(ctx context.Context,
+	projectTagToBeDelete string) (string, error) {
+	script := `
+		if (ctx._source['projects'] != null) {
+			def foundIndex = -1;
+			for (def index = 0; index < ctx._source.projects.size(); index++) {
+				if (ctx._source.projects[index] == params.project) {
+					foundIndex = index;
+					break;
+				}
+			}
```

Comment:
I think this might do your lookup more concisely (assuming projects is an `ArrayList`)--it also returns `-1` if not found:
```
def foundIndex = ctx._source.projects.indexOf(params.project)
```
Reference: https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#indexOf-java.lang.Object-

(Same comment for nodes.go below)

Reply:
Thank you, I will try it out.